src/main/java/io/cdap/plugin/salesforce/plugin/source/batch/util/BulkConnectionRetryWrapper.java

```diff
@@ -59,30 +59,45 @@ public JobInfo createJob(JobInfo jobInfo) throws AsyncApiException {
     if (!retryOnBackendError) {
       return bulkConnection.createJob(jobInfo);
     }
-    Object resultJobInfo = Failsafe.with(retryPolicy).onFailure(event -> LOG.info("Failed while creating job."))
-        .get(() -> {
-          try {
-            return bulkConnection.createJob(jobInfo);
-          } catch (AsyncApiException e) {
-            throw new SalesforceQueryExecutionException(e);
-          }
-        });
+    Object resultJobInfo;
+    try {
+      resultJobInfo = Failsafe.with(retryPolicy).onFailure(event -> LOG.info("Failed while creating job.")).get(() -> {
+        try {
+          return bulkConnection.createJob(jobInfo);
+        } catch (AsyncApiException e) {
+          throw new SalesforceQueryExecutionException(e);
+        }
+      });
+    } catch (FailsafeException fse) {
+      if (unwrapFailsafeException(fse) instanceof AsyncApiException) {
+        throw (AsyncApiException) unwrapFailsafeException(fse);
```

Comment:
I am unable to understand if we are wrapping `AsyncApiException` with `SalesforceQueryExecutionException`, why do we require this?

Shouldn't we be catching `SalesforceQueryExecutionException`?

Reply:
In this particular code, I only see bulk connection calls, which means we are catching all `AsyncApiException`, am I missing something?